include/linux/blkdev_compat.h

```diff
@@ -311,10 +311,14 @@ bio_set_flags_failfast(struct block_device *bdev, int *flags)
  * the block layers responsibility to choose the correct way to
  * implement these semantics.
  */
-#ifdef WRITE_FLUSH_FUA
+#if defined(WRITE_FLUSH_FUA)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_FLUSH_FUA
-#else
+#elif defined(WRITE_BARRIER)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_BARRIER
+#elif defined(HAVE_REQ_FUA)
+#define	VDEV_WRITE_FLUSH_FUA		REQ_FUA
```

Comment:
Actually, I think was wrong, we should use `REQ_PREFLUSH` instead of `REQ_FUA`. `REQ_FUA` only by pass cache for this write on device that supports it.